include/robot_localization/navsat_transform.hpp

```diff
@@ -57,7 +57,7 @@ class NavSatTransform
   /**
    * @brief Constructor
    */
-  NavSatTransform();
+  NavSatTransform(std::shared_ptr<rclcpp::Node> node);
```

Comment:
Best practices for this is that the NavSatTransform node should derive from rclcpp::Node, and when you make the shared_ptr of it in `main` that will deal with this, then all of your calls are working with methods available vs storing your `node_`. The ros nodehandle methodology is depreciated in ros2, its preferred to derive from the node to best make use of lifecycle and components, which is important for this package in particular

Reply:
I actually tried to use that convention here as well, but I couldn't get it to work, always got the bad_weak_ptr exception during runtime because of the TF2 package (I think). The initializer lists wants a shared, and since there is now owner when it's passed I got that exception. However it could have been the way I used image_transport the "old" way, instead of the `image_transport::create_subscription/create_publisher` that made it fail, I can't fully recall I'll need to retest it.

Reply:
If you `make_shared` the NavSatTransform object in `main` that will fix that. You get an issue with shared_ptrs if there's not already the reference counter set